infra/scoped/api-gateway.tf

```diff
@@ -295,6 +295,176 @@ resource "aws_api_gateway_method_response" "users_userid_put_500" {
   }
 }
 
+# /users/:user_id/registration
+resource "aws_api_gateway_resource" "users_userid_registration" {
+  rest_api_id = aws_api_gateway_rest_api.identity.id
+  parent_id   = aws_api_gateway_resource.users_userid.id
+  path_part   = "registration"
+}
+
+# [OPTIONS]
+
+resource "aws_api_gateway_method" "users_userid_registration_options" {
+  rest_api_id   = aws_api_gateway_rest_api.identity.id
+  resource_id   = aws_api_gateway_resource.users_userid_registration.id
+  http_method   = "OPTIONS"
+  authorization = "NONE"
+}
+
+# 204 No Content
+
+resource "aws_api_gateway_method_response" "users_userid_registration_options_204" {
+  rest_api_id = aws_api_gateway_rest_api.identity.id
+  resource_id = aws_api_gateway_resource.users_userid_registration.id
+  http_method = aws_api_gateway_method.users_userid_registration_options.http_method
+  status_code = "204"
+
+  response_models = {
+    "application/json" = "Empty"
+  }
+
+  response_parameters = {
+    "method.response.header.Access-Control-Allow-Headers" = true,
+    "method.response.header.Access-Control-Allow-Methods" = true,
+    "method.response.header.Access-Control-Allow-Origin"  = true
+  }
+}
+
+# [PUT]
+
+resource "aws_api_gateway_method" "users_userid_registration_put" {
+  rest_api_id          = aws_api_gateway_rest_api.identity.id
+  resource_id          = aws_api_gateway_resource.users_userid_registration.id
+  http_method          = "PUT"
+  authorization        = "CUSTOM"
+  authorizer_id        = aws_api_gateway_authorizer.token_authorizer.id
+  api_key_required     = true
+  request_validator_id = aws_api_gateway_request_validator.full.id
+
+  request_parameters = {
+    "method.request.path.userId" = true
+  }
+}
+
+# 200 OK
+
+resource "aws_api_gateway_method_response" "users_userid_registration_put_200" {
+  rest_api_id = aws_api_gateway_rest_api.identity.id
+  resource_id = aws_api_gateway_resource.users_userid_registration.id
+  http_method = aws_api_gateway_method.users_userid_registration_put.http_method
+  status_code = "200"
+
+  response_models = {
+    "application/json" = "Empty"
+  }
+
+  response_parameters = {
+    "method.response.header.Access-Control-Allow-Origin" = true
+  }
+}
+
+# 400 Bad Request
+
+resource "aws_api_gateway_method_response" "users_userid_registration_put_400" {
```

Comment:
We have modules for configuring these error responses; see https://github.com/wellcomecollection/terraform-aws-api-gateway-responses

In particular, they'll set up the JSON error responses we use in the rest of our APIs, e.g.

```json
{
  "errorType": "http",
  "httpStatus": "400",
  "label": "Bad Request",
  "description": "You didn't combobulate the widget wrangler",
  "type":"Error"
}
```

For an internal-only API this is likely overkill and this is fine as-is, but I mention it for information.

Reply:
fwiw - I probably wouldn't use that here, because it's not used for the rest of this APIGW setup (this is a biiiig file, which isn't great but it _is_ easy to copy and paste - and delete). Better would be to refactor the whole stack to be more modular in a separate piece of work.

Reply:
Yup, agreed – consistency within this stack is better than using this module.